src/selectors/functions/builtInFunctions.fontoxpath.js

```diff
@@ -0,0 +1,59 @@
+import DynamicContext from '../DynamicContext';
+import Sequence from '../dataTypes/Sequence';
+
+/**
+ * @param  {../DynamicContext}      dynamicContext
+ * @param  {../dataTypes/Sequence}  query
+ * @param  {../dataTypes/Sequence}  args
+ */
+function fontoxpathEvaluate (dynamicContext, query, args) {
+	/**
+	 * @type {string}
+	 */
+	const queryString = query.first().value;
+	const variables = {};
+	if (args) {
+		args.first().keyValuePairs.reduce((expandedArgs, arg) => {
+			expandedArgs[arg.key.value] = arg.value;
+			return expandedArgs;
+		}, variables);
+	}
+
+	const selector = dynamicContext.createSelectorFromXPath(queryString);
+	// Take of the context item
+	const contextItemSequence = args['.'] || Sequence.empty();
+	delete args['.'];
+	const context = contextItemSequence.isEmpty() ? {
+		contextItem: null,
+		contextSequence: contextItemSequence,
+		contextItemIndex: -1,
+		variables,
+		domFacade: dynamicContext.domFacade,
+		resolveNamespacePrefix: dynamicContext.resolveNamespacePrefix,
+		createSelectorFromXPath: dynamicContext.createSelectorFromXPath
```

Comment:
These properties always just seem to be passed on. You could consider a factory function which can build one DynamicContext from another, or adding an intermediate level to avoid having to update all context-constructing code for future additions.